### PR TITLE
[PLT-8186] add support for ec2 instance profile authentication

### PR DESCRIPTION
**Pull request by cpanato**
#### Summary
 add support for ec2 instance profile authentication

#### Ticket Link
github: https://github.com/mattermost/mattermost-server/issues/7875

#### PR in other repos
server: https://github.com/mattermost/mattermost-server/pull/8243
webapp: https://github.com/mattermost/mattermost-webapp/pull/772
api-reference: https://github.com/mattermost/mattermost-api-reference/pull/336

### CODE REVIEW

File: src/actions/admin.js [hunk 1]
```diff
@@ -70,6 +70,16 @@ export function testEmail(config) {
...
 
+export function testS3Connection(config) {
+    return bindClientFunc(
+        Client4.testS3Connection,
+        AdminTypes.TEST_S3_REQUEST,
+        AdminTypes.TEST_S3_SUCCESS,
+        AdminTypes.TEST_S3_FAILURE,
+        config
+    );
+}
```
Comment (jwilander) [inline src/actions/admin.js]: Oh sorry, can you add a quick unit test for this action?

Comment (cpanato) [reply src/actions/admin.js]: added test, please let me know if that is correct. thanks @jwilander

**Comment by jwilander** [comment]
Will merge when server PR goes in

**Comment by cpanato** [comment]
@jwilander shall we make the same changes to the test email?

**Comment by jasonblais** [comment]
@jwilander @enahum @cpanato Server PR is in, so this can now be merged

**Pull request merged**